src/plugins/vis_type_table/public/components/table_vis_component.tsx

```diff
@@ -92,22 +88,22 @@ export const TableVisComponent = ({
 
   const onColumnResize: EuiDataGridProps['onColumnResize'] = useCallback(
     ({ columnId, width }) => {
-      const prevState: ColumnWidth[] = currentColState.current.columnsWidth;
+      const curState: ColumnWidth[] = handlers.uiState.get('vis.columnsWidth') || [];
+      const nextState = [...curState];
       const nextColIndex = columns.findIndex((col) => col.id === columnId);
-      const prevColIndex = prevState.findIndex((col) => col.colIndex === nextColIndex);
-      const nextState = [...prevState];
-      const updatedColWidth = { colIndex: nextColIndex, width };
+      const curColIndex = curState.findIndex((col) => col.colIndex === nextColIndex);
+      const nextColWidth = { colIndex: nextColIndex, width };
 
       // if updated column index is not found, then add it to nextState
       // else reset it in nextState
-      if (prevColIndex < 0) nextState.push(updatedColWidth);
-      else nextState[prevColIndex] = updatedColWidth;
+      if (curColIndex < 0) nextState.push(nextColWidth);
+      else nextState[curColIndex] = nextColWidth;
 
       // update uiState
-      currentColState.current.columnsWidth = nextState;
       handlers.uiState.set('vis.columnsWidth', nextState);
+      handlers.uiState?.emit('reload');
```

Comment:
Do we need some error handling for the case that `handlers.uiState?.emit` is `undefined`? Maybe it's something we want to fail earlier on or else wait for?

Reply:
I will remove the question mark and open a follow-up issue for error handling.

Reply:
I update and add an item in the overall follow-up issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2551